From 10a887600b5df224db4adc332c691f6a729517ee Mon Sep 17 00:00:00 2001
From: Yu Xia <xiax.yu@intel.com>
Date: Mon, 25 Apr 2016 16:10:11 +0000
Subject: [PATCH 1/4] input: touch: Firmware Memory should be free after have
 used Firmware. GSL_CONFIG_DATA_ID do not need init twice.

Change-Id: Ib1649505850566045cc2da59acf4f47e5dddf2f1
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-30276
Signed-off-by: Yu Xia <xiax.yu@intel.com>
Reviewed-on: https://android.intel.com:443/498978
Signed-off-by: Yu Xia <xiax.yu@intel.com>
---
 drivers/input/touchscreen/silead.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/input/touchscreen/silead.c b/drivers/input/touchscreen/silead.c
index 1ea7b00..3bdf036 100644
--- a/drivers/input/touchscreen/silead.c
+++ b/drivers/input/touchscreen/silead.c
@@ -373,6 +373,9 @@ static int silead_ts_load_fw(struct i2c_client *client)
 	return 0;
 
 release_fw_err:
+	release_firmware(fw);
+	fw = NULL;
+
 	return ret;
 }
 
@@ -707,6 +710,7 @@ static int silead_ts_probe(struct i2c_client *client,
 					client->name, data);
 	if (ret) {
 		dev_err(dev, "IRQ request failed %d\n", ret);
+		input_unregister_device(data->input_dev);
 		return ret;
 	}
 
@@ -723,9 +727,6 @@ static int silead_ts_probe(struct i2c_client *client,
 out:
 #endif
 
-#ifdef GSL_ALG_ID
-	gsl_DataInit(gsl_config_data_id);
-#endif
 	dev_dbg(dev, "Probing succeded\n");
 	return 0;
 }
-- 
1.9.1

